test/utils.ts

```diff
@@ -154,3 +156,58 @@ export const signOrder = async (signer: any, order: any) => {
   order.signature = `0x${v}${rs}02`
   return order
 }
+
+export const expectedLiqudidationPrice = (
+  collateral: number | string,
+  debt: number,
+  cashValue: number,
+  spotPrice: number,
+  oracleDeviation: number,
+  auctionStartingTime: number,
+  currentBlockTime: number,
+  isPut: boolean,
+  collateralDecimals: number,
+) => {
+  const endingPrice = new BigNumber(collateral).dividedBy(debt)
+  const auctionElapsedTime = currentBlockTime - auctionStartingTime
+
+  if (auctionElapsedTime > 3600) {
+    // return Math.floor(endingPrice)
+    return endingPrice.multipliedBy(10 ** collateralDecimals).toNumber()
+  }
+
+  let startingPrice
+
+  if (isPut) {
+    startingPrice = BigNumber.max(
+      new BigNumber(cashValue).minus(new BigNumber(spotPrice).multipliedBy(oracleDeviation)),
+      0,
+    )
+  } else {
+    startingPrice = BigNumber.max(
+      new BigNumber(cashValue).minus(new BigNumber(spotPrice).multipliedBy(oracleDeviation)),
+      0,
+    ).dividedBy(spotPrice)
+  }
+
+  return startingPrice
+    .plus(
+      endingPrice
+        .minus(startingPrice)
+        .multipliedBy(auctionElapsedTime)
+        .dividedBy(3600),
+    )
+    .multipliedBy(10 ** collateralDecimals)
+    .toNumber()
+}
+
+export const calcRelativeDiff = (expected: BigNumber, actual: BigNumber): BigNumber => {
+  let diff: BigNumber
+
+  if (actual.isGreaterThan(expected)) {
+    diff = actual.minus(expected)
+  } else {
+    diff = expected.minus(actual)
+  }
+  return diff
```

Comment:
I think there's an `.abs()` function, so you can use `actual.minus(expected).abs()`